### PR TITLE
bugfix: throw error meg when call admin API to enable mqtt plugin but `stream_proxy` is disabled.

### LINKED ISSUE #943: bug:  not throw error meg when call admin API to enable mqtt plugin and `enable stream_proxy` is disable.
as discussed here: https://github.com/apache/incubator-apisix/issues/899#issuecomment-559626538

### PR DESCRIPTION
**Pull request by membphis**
fix #943 .

### CODE REVIEW

File: t/admin/stream-routes.t [hunk 1]
```diff
@@ -15,13 +15,42 @@
...
+    my $data = do { local $/; <$in> };
+    close $in;
+    $data;
+}
+
+my $yaml_config = read_file("conf/config.yaml");
+$yaml_config =~ s/node_listen: 9080/node_listen: 1984/;
+$yaml_config =~ s/enable_heartbeat: true/enable_heartbeat: false/;
+$yaml_config =~ s/  # stream_proxy:/  stream_proxy:\n    tcp:\n      - 9100/;
+$yaml_config =~ s/admin_key:/admin_key_useless:/;
```
Comment (moonming) [inline t/admin/stream-routes.t:41]: what is `admin_key_useless`?

File: apisix/admin/init.lua [hunk 2]
```diff
@@ -147,6 +147,74 @@ local function run()
 end
 
 
+local function run_stream()
+    local api_ctx = {}
+    core.ctx.set_vars_meta(api_ctx)
+
+    local local_conf = core.config.local_conf()
+    if not local_conf.apisix.stream_proxy then
+        core.log.warn("disabled stream model, can not to add any stream route")
```
Comment (moonming) [inline apisix/admin/init.lua:156]: `disabled stream model` -> `The stream mode is disabled.`

File: t/admin/stream-routes.t (hunk 1 above)
Comment (membphis) [reply t/admin/stream-routes.t:41]: One way to disable admin API token

https://github.com/apache/incubator-apisix/blob/master/t/APISIX.pm#L77

Comment (moonming) [reply t/admin/stream-routes.t:41]: I don't know what that means in English：admin_key_useless

Comment (membphis) [reply t/admin/stream-routes.t:41]: how about `disable_admin_key` ?

**Comment by moonming** [comment]
lgtm

Thanks,
Ming Wen
Twitter: _WenMing


YuanSheng Wang <notifications@github.com> 于2020年4月7日周二 上午11:08写道：

**Pull request merged**